crates/node/src/workflow/mod.rs

```diff
@@ -82,7 +82,7 @@ impl WorkflowEngine {
 
                 match workflow.steps.iter().position(|s| s.program == *prover) {
                     Some(proof_step_idx) => {
-                        if workflow.steps.len() <= proof_step_idx {
+                        if workflow.steps.len() <= (proof_step_idx + 1) {
```

Comment:
Under stress test I faced a situation where `workflow.steps.len()` was `2` and `proof_step_idx` was `1`. This obviously then crashed on line 95...